### PR TITLE
FW-1563 Songs and Stories Mock Data

### PR DESCRIPTION
**Pull request by elysse-fpcc**
- Added code to generate 5 songs and 5 stories of variable lengths
- Each song/story has randomly generated title/introduction/page content of varying sizes
- Songs have between 0-5 pages, stories have 4-15 pages
- Also added a new document "contributor" for use in Songs/Stories as well as in future improvements on mock data 
- Updated tests to test for these new document types

### CODE REVIEW

File: FirstVoicesCoreTests/src/main/java/ca/firstvoices/tests/mocks/services/MockDialectServiceImpl.java [hunk 1]
```diff
@@ -386,4 +407,137 @@ private DocumentModelList generateFVPhraseBooks(CoreSession session, String path
...
+      Map<String, String> introductionTranslationEntry = new HashMap<>();
+      introductionTranslationEntry.put(TRANSLATION, "Introduction translation to English");
+      introductionTranslationEntry.put(LANGUAGE, ENGLISH);
+      introductionTranslation.add(introductionTranslationEntry);
+      songDoc.setPropertyValue("fvbook:introduction_literal_translation", introductionTranslation);
+
+      songDoc.setPropertyValue("fvbook:acknowledgement", "Acknowledgement");
+      String[] culturalNoteArr = {"Cultural note"};
+      songDoc.setPropertyValue("fv:cultural_note", culturalNoteArr);
+      String[] contributorsArray = {"Contributor"};
```
Comment (dy2288) [inline FirstVoicesCoreTests/src/main/java/ca/firstvoices/tests/mocks/services/MockDialectServiceImpl.java:453]: Don't think setting `dc:contributors` is needed here?

File: FirstVoicesCoreTests/src/main/java/ca/firstvoices/tests/mocks/services/MockDialectServiceImpl.java [hunk 2]
```diff
@@ -386,4 +407,137 @@ private DocumentModelList generateFVPhraseBooks(CoreSession session, String path
...
+      songDoc.setPropertyValue("dc:contributors", contributorsArray);
+
+      fvSongs.add(createDocument(session, songDoc));
+
+      // generate pages within song
+      String songPath = songDoc.getPathAsString();
+      int numPages = ThreadLocalRandom.current().nextInt(0, 5);
```
Comment (dy2288) [inline FirstVoicesCoreTests/src/main/java/ca/firstvoices/tests/mocks/services/MockDialectServiceImpl.java:460]: For the purposes of setting mock data for something like cypress, it probably makes sense to always have at least one page generated?

File: FirstVoicesCoreTests/src/main/java/ca/firstvoices/tests/mocks/services/MockDialectServiceImpl.java [hunk 3]
```diff
@@ -386,4 +407,137 @@ private DocumentModelList generateFVPhraseBooks(CoreSession session, String path
...
+            .createDocumentModel(songPath, pageTitle, FV_BOOK_ENTRY);
+        pageDoc.setPropertyValue("fvbookentry:dominant_language_text", pageTranslation);
+        fvPage.add(createDocument(session, pageDoc));
+      }
+    }
+
+    return fvSongs;
+  }
+
+  private DocumentModelList generateFVStories(CoreSession session, String path, int storyEntries,
```
Comment (dy2288) [inline FirstVoicesCoreTests/src/main/java/ca/firstvoices/tests/mocks/services/MockDialectServiceImpl.java:482]: Aside from the line that sets the type and the variable names (e.g. `storyDoc.setPropertyValue("fvbook:type", "story");`) - this and the generateFVSongs() methods are pretty much identical. I would avoid the duplication by creating a generic method to handle the creation of a book, that can accept the type (song/story).

Then call it in here, e.g.:
```java
private DocumentModelList generateFVStories(CoreSession session, String path, int storyEntries,
String[] wordsToUse) {
   createBook(session, 'story');
}
```

and 

```java
private DocumentModelList generateFVSongs(CoreSession session, String path, int storyEntries,
String[] wordsToUse) {
   createBook(session, 'song');
}
```

**Review by dy2288 — CHANGES_REQUESTED** [review]
Great job overall @elysse-fpcc - see proposed changes though.

**Comment by elysse-fpcc** [comment]
Thanks for the review and feedback @dy2288 - made the requested changes :)

**Review by dy2288 — APPROVED** [review]
Looks great! Approved

**Pull request merged**